src/authorization/repository/permission.repository.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import { DataSource } from 'typeorm';
+import Permission from '../entity/permission.entity';
+import { BaseRepository } from './base.repository';
+import { GroupPermissionRepository } from './grouppermission.repository';
+import { UserPermissionRepository } from './userpermission.repository';
+import {
+  NewPermissionInput,
+  UpdatePermissionInput,
+} from 'src/schema/graphql.schema';
+import {
+  PermissionDeleteNotAllowedException,
+  PermissionNotFoundException,
+} from '../exception/permission.exception';
+
+@Injectable()
+export class PermissionRepository extends BaseRepository<Permission> {
+  constructor(
+    private dataSource: DataSource,
+    private userPermissionRepository: UserPermissionRepository,
+    private groupPermissionRepository: GroupPermissionRepository,
+  ) {
+    super(Permission, dataSource);
+  }
+
+  getAllPermissions(): Promise<Permission[]> {
+    return this.find();
+  }
+
+  async getPermissionById(id: string): Promise<Permission> {
```

Comment:
This method should only contain the queries. Any logic should be placed in the service layer itself.

Reply:
Thanks, have made the necessary changes.